lib/manageiq/api/common/middleware/web_server_metrics.rb

```diff
@@ -0,0 +1,53 @@
+module ManageIQ
+  module API
+    module Common
+      module Middleware
+        class WebServerMetrics
+          def initialize(app, options = {})
+            @app            = app
+            @metrics_prefix = options[:metrics_prefix] || "http_server"
+
+            require 'prometheus_exporter/client'
+            require 'prometheus_exporter/metric'
+
+            @counter   = PrometheusExporter::Client.default.register(:counter, "#{@metrics_prefix}_total", "The total number of HTTP requests handled by the Rack application.")
+            @gauge     = PrometheusExporter::Client.default.register(:gauge, "#{@metrics_prefix}_busy_threads", "The number of threads currently handling HTTP requests.")
+            @histogram = PrometheusExporter::Client.default.register(:histogram, @metrics_prefix, "The HTTP response duration of the Rack application.")
```

Comment:
This metric should have a better name. Try to comply with the best practices defined in the prometheus documentation https://prometheus.io/docs/practices/naming/

Also I think the existing API metrics are a good example:
![image](https://user-images.githubusercontent.com/12697904/58968848-b72b1e80-8784-11e9-84e4-ef7cb2379798.png)